articles/quickstart/native/ios-swift-siwa/_includes/_login_centralized.md

```diff
@@ -0,0 +1,306 @@
+
+```swift
+extension ViewController: ASAuthorizationControllerDelegate {
+}
+```
+
+### Add the Sign In With Apple capability
+
+To enable Sign In With Apple in your application, you need to enable this capability to your app, as shown:
+
+![Add capabilities](/media/articles/ios/swift/add-capability.png)
+
+:::panel Checkpoint
+Start the application and click the **Sign In With Apple** button. You should see the default sign-in dialog appear:
+
+![Default sign-in dialog](/media/articles/ios/swift/authz-dialog.png)
+:::
+
+## Process the Authorization Response
+
+Add the following to the end of your `ViewController.swift` file.
+
+```swift
+extension ViewController: ASAuthorizationControllerDelegate {
+    // Handle authorization success
+    func authorizationController(controller: ASAuthorizationController, didCompleteWithAuthorization authorization: ASAuthorization) {
+
+        if let appleIDCredential = authorization.credential as? ASAuthorizationAppleIDCredential {
+            // Success
+        }
+    }
+
+    // Handle authorization failure
+    func authorizationController(controller: ASAuthorizationController, didCompleteWithError error: Error) {
+        print("Authorization Failed: \(error)")
+    }
+}
+```
+
+### Inspect the response
+
+Upon successful authorization, you will be able to access the following information:
+
+- UUID, team-scoped user ID
+- Verification Data
+  - ID Token, Authorization Code
+- Account Information
+  - Name, Verified Email Address (Original or relay address)
+- Real user indicator
+  - High confidence indicator that the user is who they say they are
+
+:::note
+The Account Information is ONLY sent on the first successful authorization
+:::
+
+## Auth0 Token Exchange
+
+Now that you have a successful authorization response, you can use the `authorizationCode` information to perform a token exchange to gain access to Auth0 credentials, such as the ID and Access Tokens, as in the following example:
+
+```swift
+if let appleIDCredential = authorization.credential as? ASAuthorizationAppleIDCredential {
+    // Convert Data -> String
+    guard let authorizationCode = appleIDCredential.authorizationCode, let authCode = String(data: authorizationCode, encoding: .utf8) else
+    {
+        print("Problem with the authorizationCode")
+        return
+    }
+
+    // Auth0 Token Exchange
+    Auth0
+        .authentication()
+        .tokenExchange(withAppleAuthorizationCode: authCode).start { result in
+            switch(result) {
+            case .success(let credentials):
+                print("Auth0 Success: \(credentials)")
+
+            case .failure(let error):
+                print("Exchange Failed: \(error)")
+            }
+
+    }
+}
+```
+
+## Renew Authentication
+
+When users return to your app, it would be beneficial to log them in automatically provided their login session is still valid. This involves:
```

Comment:
When users return to your app, you can log them in automatically if their login session is still valid. This involves: